src/util/color.ts

```diff
@@ -30,59 +32,88 @@ class Color {
-    static parse(input?: string | Color | null): Color | void {
-        if (!input) {
-            return undefined;
+    static parse(input: String | string | undefined | null): Color | undefined {
+        if (Object.prototype.toString.call(input) !== '[object String]') {
+            return;
         }
 
-        if (input instanceof Color) {
-            return input;
+        const parsedColor = parseCssColor(input.toLowerCase());
+        if (parsedColor) {
+            const [r, g, b, alpha] = parsedColor;
+            const color = new Color(r * alpha, g * alpha, b * alpha, alpha);
+            if (!alpha) {
+                // alpha = 0 erases completely rgb channels. This behavior is not desirable
+                // if this particular color is later used in color interpolation.
+                // Because of that, a reference to original color is saved.
+                color.overwriteGetter('rgb', parsedColor);
+            }
+            return color;
         }
+    }
 
-        if (typeof input !== 'string') {
-            return undefined;
-        }
+    /**
+     * Used in color interpolation.
+     *
+     * @returns Gien color, with reversed alpha blending, in sRGB color space.
+     */
+    get rgb(): RGBColor {
+        const {r, g, b, a} = this;
+        return this.overwriteGetter('rgb', [r / a, g / a, b / a, a]);
+    }
 
-        const rgba = parseCSSColor(input);
-        if (!rgba) {
-            return undefined;
-        }
+    /**
+     * Used in color interpolation.
+     *
+     * @returns Gien color, with reversed alpha blending, in HCL color space.
+     */
+    get hcl(): HCLColor {
+        return this.overwriteGetter('hcl', hcl.fromRgb(this.rgb));
+    }
 
-        return new Color(
-            rgba[0] / 255 * rgba[3],
-            rgba[1] / 255 * rgba[3],
-            rgba[2] / 255 * rgba[3],
-            rgba[3]
-        );
+    /**
+     * Used in color interpolation.
+     *
+     * @returns Gien color, with reversed alpha blending, in LAB color space.
+     */
+    get lab(): LABColor {
+        return this.overwriteGetter('lab', lab.fromRgb(this.rgb));
     }
 
     /**
-     * Returns an RGBA string representing the color value.
+     * Lazy getter pattern. When getter is called for the first time lazy value
+     * is calculated and then overwrites getter function in given object instance.
      *
-     * @returns An RGBA string.
-     * @example
-     * var purple = new Color.parse('purple');
-     * purple.toString; // = "rgba(128,0,128,1)"
-     * var translucentGreen = new Color.parse('rgba(26, 207, 26, .73)');
-     * translucentGreen.toString(); // = "rgba(26,207,26,0.73)"
+     * @param getterKey Getter key
+     * @param lazyValue Lazy value
+     * @private
      */
-    toString(): string {
-        const [r, g, b, a] = this.toArray();
-        return `rgba(${Math.round(r)},${Math.round(g)},${Math.round(b)},${a})`;
+    private overwriteGetter<T>(getterKey: string, lazyValue: T): T {
```

Comment:
Can you elaborate on the purpose behind this method? seems complicated...

Reply:
This acts as a local lazy cache for each color instance. 
Once calculated the value of a given color in HCL/LAB color space can be reused instead of recalculated every time it is needed.
So when interpolating `red` to `blue` in HCL space, the colors `red` and `blue` are converted to HCL each time an intermediate point is calculated (i.e. `interpolate.color` is called), which can be counted in several hundreds of times for these two colors alone. Memoization of `red` and `blue` values in HCL should positively affect performance of interpolations in HCL/LAB. 
Treating sRGB values in a similar way is mainly for consistency purposes, but it serves the important role of storing rgb channel values when alpha=0.

Here you could read more about lazy getters:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get#smart_self-overwriting_lazy_getters

Reply:
Is there a more readable way to write this? while I understand the code after you explain it, I don't expect someone to understand this and this can cause problems.
Also, is there a problem we know this solves? I don't like premature optimizations...

Reply:
This is implementation details for the `Color` class. Exposed for `Color` class users are 3 getters: `rgb`, `hcl` and `lab` each with a specified type.
However, if at some point someone will want to understand how this internal function works, specially for him/her I expanded jsdoc with a comprehensive and illustrative example that should clear all uncertainties.